### ISSUE TITLE
Diff view for content revisions

### ISSUE THREAD

**Issue by pspiller** [unite-cms/unite-cms#669]
At the moment, it can be a bit hard to tell what's changed between content revisions. A diff view would be very useful here - you could select a revision range for a piece of content, then see a highlighted view of the differences. Is anything like this on the roadmap? Would you be interested in a PR for this feature?

**Comment by franzwilding** [comment]
Yes, that would be cool. We already include a diff-tool for domain update changes (https://github.com/unite-cms/unite-cms/blob/master/src/Bundle/CoreBundle/Resources/views/Domain/_change_visualization.html.twig https://github.com/unite-cms/unite-cms/blob/master/src/Bundle/CoreBundle/Resources/webpack/main.js#L127) so you would just need to implement a better revisions view. 

A PR would be awesome!

**Comment by pspiller** [comment]
Our ideas on how we'd update the content revisions interface:

1. Add a 'Revision description' field to the content edit page, on the same row as the 'Submit' button. This would let users enter a description of the changes they're making to the content item.

2. Add a new optional `revision_description_required` field to content types in the domain configuration. If present and set to `true`, then users would have to provide a revision description before they can save an entity of that content type.

3. Don't show the content for each revision inline, as the page can get very long for frequently-updated pieces of content. Instead, just show the version number, description, action, timestamp and actor fields in a tabular view.

4. Add an interface to allow selecting a range of revisions. An example would be the UI from a tool called Phabricator: two radio button columns. This would let you select a range of revisions to view a diff between, or a single revision to view the content for.

![phab_history_selector](https://user-images.githubusercontent.com/1845477/60692620-44939700-9f1a-11e9-8373-2c81d3ecf0b3.png)

5. Once a revision range is selected, add a button that lets you view a diff between the two versions. As you mentioned, we'd re-use the Ace editor diff mode for this. If a single revision was selected in both columns, you'd just show that rather than a diff.

Any comments, or does that sound reasonable?

**Comment by franzwilding** [comment]
Sounds very powerful. Some thoughts about your idea:

1. a Revision comment field must be optional, in many cases you don‘t want to show this field at all. Maybe we could have a normal field for that and a setting, to use this as the revision description field?

2. Maybe we find a nicer UI for the revisions overview page. A use case for me would be to see an overview of small changes (for example a status field that get changed a lot). In this case, it would be nice to easily see a good overview of the changes. For larger change-sets I totally agree, that something like your proposed UI would be good

**Comment by franzwilding** [comment]
With version 0.10 the revisions view is a table view where the user can see the differences for each field per version (row). A diff view would be a cool extra, however I think the table view will be enough for version 1.0

**Comment by its-lucas** [comment]
Using the table view in 0.10 I'm having trouble with viewing changes to collections and variants. The below image shows the table view for the categories type where I've updated the adminView fragment to show the collection fields since I would need to do this in a production environment.
 
![image](https://user-images.githubusercontent.com/60761410/74397941-07a56080-4e6a-11ea-963c-cbedb7919e36.png)

The fragment used can be found at: https://gist.github.com/its-lucas/fc9e7b70b723a80324403c8d9b82082d

From what I understand, it is similar to the content views in 0.9, in that you have to define which fields will be visible in the view, and in the case of a content revisions interface, the only way you could know which fields have had their values updated is if you always explicitly add every field to the fragment for all content types.

I would be much more in favour of this proposed diff view if possible?
